content/outcomes/secure-software-supply-chain/building-a-container.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Building a Container
+weight: 2
+layout: single
+---
+
+There's many options that you have when deciding how you want to build the container for your code. From writing a Dockerfile by hand to automated build services, finding the right build method is an important first step. Let's take a look at a few options that exist do see the advantages and disadvantages of each!
+
+## Dockerfile
+
+If you're already familiar with building a container image, this is probably where you got your start. A Dockerfile defines a set of steps to build you container, as well as if it should use another container image as a base. This allows us to reuse more generic container images that may contain things such as the appropriate application runtimes and dependencies. Consider this basic Dockerfile:
+
+```dockerfile
+FROM ubuntu:18.04
+COPY . /app
+RUN make /app
+ENTRYPOINT /app/my-apo
+```
+
+This Dockerfile tells the builder to use the `ubuntu:18.04` container image as a base and to copy the current directory from the local filesystem to `/app` within the container that we're building. The `RUN` command then runs the command to build the application. Finally, the `ENTRYPOINT` instruction defines what to run when the container is started. In this case, when the container is started, the executable at `/app/my-app` is ran.
```

Comment:
tells the builder => tells the Docker build process

Reply:
within the container that => within the filesystem of the container

Reply:
to build the application => to build the application within the context of the container image.

Reply:
defines what to run => defines the command to run

is ran => is launched by default. Users may override this entrypoint on the command line or through Kubernetes Pod specifications.